src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -64,21 +64,28 @@
 VERSION_FILE = 'versions.yaml'
 COMPUTE_TOOLS_DIR = '../../compute_tools'  # relative to task_generator.py
 
-# TODO GTDB update readme to specify to get correct version of data, not just latest
-# TODO REPRODUCIBILITY need to version the databases and use the correct version with the
-#                      correct tool version
-# directory containing the unarchived GTDB-Tk reference data
-# download data following the instructions provided on
-# https://ecogenomics.github.io/GTDBTk/installing/index.html#gtdb-tk-reference-data
-GTDBTK_DATA_PATH = '/global/cfs/cdirs/kbase/collections/libraries/gtdb_tk/release207_v2'
+# volume name for the Docker containers
+TOOL_IMG_VOLUME_NAME = {'checkm2': '/CheckM2_database',
+                        'gtdb_tk': '/gtdbtk_reference_data'}
+LIBRARY_DIR = 'libraries'  # subdirectory for the library files
 
-# DIAMOND database CheckM2 relies on
-# download data following the instructions provided on https://github.com/chklovski/CheckM2#database
-CHECKM2_DB = '/global/cfs/cdirs/kbase/collections/libraries/CheckM2_database'
 
-# volume mapping for the Docker containers
-TOOL_VOLUME_MAP = {'checkm2': {CHECKM2_DB: '/CheckM2_database'},
-                   'gtdb_tk': {GTDBTK_DATA_PATH: '/gtdbtk_reference_data'}}
+def _retrieve_tool_volume(tool, root_dir):
+    # Retrieve the volume mapping for the specified tool.
+
+    current_dir = os.path.dirname(os.path.abspath(__file__))
+    compute_tools_dir = os.path.join(current_dir, COMPUTE_TOOLS_DIR)
+    version_file = os.path.join(compute_tools_dir, tool, VERSION_FILE)
+    ref_db_path = extract_latest_reference_db_path(version_file)
```

Comment:
Move this block under the `if`? No need to do it if there's no mount needed

Reply:
👍

Reply:
:+1: